### PR TITLE
Update List Workers Repo function to allow limits and provide scopes

### PR DESCRIPTION
**Pull request by talanknight**

### CODE REVIEW

File: internal/servers/repository_worker.go [hunk 1]
```diff
@@ -114,39 +114,45 @@ func lookupWorker(ctx context.Context, reader db.Reader, id string) (*Worker, er
...
-
-// listWorkersWithReader will return a listing of resources and honor the
-// WithLimit option. If WithLiveness is zero the default liveness value is used,
-// if it is negative then the last status update time is ignored. This method
-// accepts a reader, allowing it to be used within a transaction or without.
-func (r *Repository) listWorkersWithReader(ctx context.Context, reader db.Reader, opt ...Option) ([]*Worker, error) {
-	const op = "workers.listWorkersWithReader"
+// ListWorkers will return a listing of Workers and honor the WithLimit option.
+// If WithLiveness is zero the default liveness value is used, if it is negative
+// then the last status update time is ignored.
```
Comment (irenarindos) [inline internal/servers/repository_worker.go:119]:
```suggestion
// then the last status update time is ignored.
// If WithLimit < 0, then unlimited results are returned. If WithLimit == 0, then 
// default limits are used for results.
```

**Review by irenarindos — APPROVED** [review]
Doc suggestion but otherwise looks good

File: internal/servers/repository_worker.go [hunk 2]
```diff
@@ -114,39 +114,47 @@ func lookupWorker(ctx context.Context, reader db.Reader, id string) (*Worker, er
...
+// If WithLimit < 0, then unlimited results are returned. If WithLimit == 0, then 
+// default limits are used for results.
+func (r *Repository) ListWorkers(ctx context.Context, scopeIds []string, opt ...Option) ([]*Worker, error) {
+	const op = "workers.(Repository).ListWorkers"
```
Comment (mgaffney) [inline internal/servers/repository_worker.go:123]:
```suggestion
	const op = "servers.(Repository).ListWorkers"
```

**Review by mgaffney — COMMENTED** [review]
One minor suggestion

**Pull request merged**